pkg/operator/ceph/controller/spec.go

```diff
@@ -624,3 +634,84 @@ func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Contai
 		Resources:       cephv1.GetLogCollectorResources(c.Resources),
 	}
 }
+
+// CreateExternalMetricsEndpoints creates external metric endpoint
+func createExternalMetricsEndpoints(namespace string, monitoringSpec cephv1.MonitoringSpec, ownerInfo *k8sutil.OwnerInfo) (*v1.Endpoints, error) {
+	labels := AppLabels("rook-ceph-mgr", namespace)
+
+	endpoints := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ExternalMgrAppName,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Subsets: []v1.EndpointSubset{
+			{
+				Addresses: monitoringSpec.ExternalMgrEndpoints,
+				Ports: []v1.EndpointPort{
+					{
+						Name:     ServiceExternalMetricName,
+						Port:     int32(monitoringSpec.ExternalMgrPrometheusPort),
+						Protocol: v1.ProtocolTCP,
+					},
+				},
+			},
+		},
+	}
+
+	err := ownerInfo.SetControllerReference(endpoints)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to set owner reference to metric endpoints %q", endpoints.Name)
+	}
+
+	return endpoints, nil
+}
+
+func ConfigureExternalMetricsEndpoint(ctx *clusterd.Context, monitoringSpec cephv1.MonitoringSpec, clusterInfo *client.ClusterInfo, ownerInfo *k8sutil.OwnerInfo) error {
+	// Get active mgr
+	var activeMgrAddr string
+	// We use mgr dump and not stat because we want the IP address
+	mgrMap, err := client.CephMgrMap(ctx, clusterInfo)
```

Comment:
When testing for stretch cluster I found "ceph mgr dump" was very verbose (~250K). I was glad to find at least in Pacific we could start calling "ceph mgr stat". Could we first detect with "stat" if the mgr changed, then fall back to "dump" only when needed? Maybe it's not that big of a deal, but if we're querying it with every health check (45-60s) it feels frequent for a big response like that. wdyt?

Reply:
> Could the mgr ip change even if the active mgr doesn't change?

Well yes, the standby IP could change, but if the active change then it will probably failover at the same time.

> If the ip only changes if the mgr is failed over, I wonder if we could call dump only if the active mgr changes.
Yes, the `active` manager IP changes if the mgr fails over.

We only compare IPs at this point, so I'm sure how the `dump`/ `stat` combination might look like without adding too much complexity.

Reply:
Yes should be easy to add.